status/reporter_github.go

```diff
@@ -314,13 +317,16 @@ func (csu *CommitStatusUpdater) createCommitStatusAdapterForSnapshot(report Test
 		return nil, fmt.Errorf("unknown status %s for integrationTestScenario %s and snapshot %s/%s", report.Status, report.ScenarioName, snapshot.Namespace, snapshot.Name)
 	}
 
+	targetURL := FormatPipelineURL(report.TestPipelineRunName, snapshot.Namespace, *csu.logger)
```

Comment:
I am thinking if we should also handle the scenario here when the `TestPipelineRunName` has not been set for CheckRun and ComitStatus? Otherwise it will produce a strange targetUrl and detailsUrl, such as for inProgres status.

Reply:
good point @hongweiliu17 , 
I will add a check on both assignments and print a proper message when **TestPipelineRunName** is not set yet .

Reply:
so we check if report.Status is in progress then we update the CheckRun  and CommitStatus with the Console ITS PLR URL , else we set the **detailsURL** and **targetURL** to **""**

Reply:
Would checking if the `report.TestPipelineRunName` is set to anything before generating a pipelineURL do the job?

Reply:
you mean like the check I've added before [detailsURL](https://github.com/kasemAlem/integration-service/blob/367141fd63a0c00ae98d69917de9c94b1fac5731/status/reporter_github.go#L182-L187)   and before the [targetURL](https://github.com/kasemAlem/integration-service/blob/367141fd63a0c00ae98d69917de9c94b1fac5731/status/reporter_github.go#L327-L332) ?

Reply:
> so we check if report.Status is in progress then we update the CheckRun and CommitStatus with the Console ITS PLR URL , else we set the **detailsURL** and **targetURL** to **""**

I am not sure if we can set **detailsURL** and **targetURL** to **""** because I don't know what will be shown if that link to **detailsURL** and **targetURL** is clicked. Or don't set it if PLR URL is ""?

Reply:
@hongweiliu17 I think that's what happens with the latest change [for detailsURL](https://github.com/redhat-appstudio/integration-service/pull/651/files#diff-9896ed3f1575fe48878f4f5d1b4d39680607d849a5fa3604fe745f0ab79efbb4R258-R260), not seeing the same for `targetURL`. I'd guess we'd need confirmation in testing on how this would look.

Reply:
> @hongweiliu17 I think that's what happens with the latest change [for detailsURL](https://github.com/redhat-appstudio/integration-service/pull/651/files#diff-9896ed3f1575fe48878f4f5d1b4d39680607d849a5fa3604fe745f0ab79efbb4R258-R260), not seeing the same for `targetURL`. I'd guess we'd need confirmation in testing on how this would look.

I agree.

Reply:
for the `targetURL` I can suggest like below 

```
func (c *Client) CreateCommitStatus(ctx context.Context, owner string, repo string, SHA string, state string, description string, statusContext string, targetURL string) (int64, error) {

	var status *ghapi.RepoStatus
	var err error

	if targetURL != "" {
		status, _, err = c.GetRepositoriesService().CreateStatus(ctx, owner, repo, SHA, &ghapi.RepoStatus{State: &state, Description: &description, Context: &statusContext, TargetURL: &targetURL})

	} else {
		status, _, err = c.GetRepositoriesService().CreateStatus(ctx, owner, repo, SHA, &ghapi.RepoStatus{State: &state, Description: &description, Context: &statusContext})
	}

	if err != nil {
		return 0, fmt.Errorf("failed to create an existing commitStatus for GitHub owner/repo/ref %s/%s/%s: %w", owner, repo, SHA, err)
	}

	c.logger.Info("Created commit status",
		"ID", status.ID,
		"Owner", owner,
		"Repository", repo,
		"SHA", SHA,
		"State", state,
		"TargetURL", targetURL,
	)

	return *status.ID, nil
}
```
@dirgim @hongweiliu17 WDYT?

Reply:
What about something like this (untested):
```
func (c *Client) CreateCommitStatus(ctx context.Context, owner string, repo string, SHA string, state string, description string, statusContext string, targetURL string) (int64, error) {
	repoStatus := ghapi.RepoStatus{}
	repoStatus.State = &state
	repoStatus.Description = &description
	repoStatus.Context = &statusContext

	if targetURL != "" {
		repoStatus.TargetURL = &targetURL
	}

	status, _, err := c.GetRepositoriesService().CreateStatus(ctx, owner, repo, SHA, &repoStatus)
	if err != nil {
		return 0, fmt.Errorf("failed to create an existing commitStatus for GitHub owner/repo/ref %s/%s/%s: %w", owner, repo, SHA, err)
	}

	c.logger.Info("Created commit status",
		"ID", status.ID,
		"Owner", owner,
		"Repository", repo,
		"SHA", SHA,
		"State", state,
		"TargetURL", targetURL,
	)
	return *status.ID, nil
}
```

Reply:
thanks @dirgim , it does looks and work better  
I would suggest more improvement on your suggestion : 
 
 instead : 
```
 	repoStatus := ghapi.RepoStatus{}
	repoStatus.State = &state
	repoStatus.Description = &description
	repoStatus.Context = &statusContext
 
```
 to : 
```
	repoStatus := ghapi.RepoStatus{
		State:       &state,
		Description: &description,
		Context:     &statusContext,
	}
```
and to add the **status** to the logger message : 

```
	c.logger.Info("Created commit status",
		"ID", status.ID,
		"Owner", owner,
		"Repository", repo,
		"SHA", SHA,
		"State", state,
		"TargetURL", targetURL,
		"Status", status, // Log the entire status object
```

Reply:
Yup, that works too.